### DISCUSSION TITLE
[self hosted] how to run seed.sql on docker compose init

### DISCUSSION THREAD

**Discussion by riccardolardi** [supabase/supabase#28045] [Questions]
During prototyping I have some SQL I'd like to run against the DB each time I re-init the project from scratch, e.g. after deleting `volumes/db/data` for clean DB state. Is there a way to have custom seed scripts that run each time the DB is re-initialized?

**Comment by silentworks** [comment]
You can add such data in this file if you want to seed the db automatically on `docker compose up`. https://github.com/supabase/supabase/blob/master/docker/volumes/db/init/data.sql You can see this is read here https://github.com/supabase/supabase/blob/master/docker/docker-compose.yml#L397-L398 in your `docker-compose.yml` file.

**Reply by riccardolardi** [reply]
Thank you!

**Reply by nicetomytyuk** [reply]
@silentworks Is the data from `/db/init/data.sql` read only on the first supabase run? Because I had a newly installed instance of supabase, did my migration things with the schemas, then tried to `docker compose down` then add my static data in `/db/init/data.sql` and run `docker compose up` again, but the data has not been initialized.

**Reply by christophe-lc** [reply]
Is this still the case? I'm trying to use this file and it doesn't seem to help to put something in there.

**Reply by nk9** [reply]
Apologies if I'm missing something, but I don't see how this 2024 answer could be correct. [The lines referred to at that time](https://github.com/supabase/supabase/blob/3980df24d29dbbadc62aa7bba9eb2c083ed107cc/docker/docker-compose.yml#L397-L398) mount `./volumes/db/data/` into the image. This mount is still there in the [current version](https://github.com/supabase/supabase/blob/d47fd2438c57a7188613d96a073f8781b7d46338/docker/docker-compose.yml#L476) of the Docker image, but the file in question is `./volumes/db/init/data.sql`, which isn't under that path.

The real solution is to add this to your `services.db.volumes` section in the docker-compose.yaml:
```
      - ./volumes/db/init/data.sql:/docker-entrypoint-initdb.d/data.sql:Z
```

You'll then see some console spew at first container startup:

```
supabase-db  | /usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initdb.d/data.sql
supabase-db  | CREATE SCHEMA
```

I'm not clear why this file even exists in the example repo if it's not used, but it doesn't seem to be.
